### PR TITLE
Automated cherry pick of #23500

### PR DESCRIPTION
**Pull request by bprashanth**
Cherry pick of #23500 on release-1.2.

### CODE REVIEW

**Comment by bprashanth** [comment]
@tam7t asking on googlebots behest (https://github.com/kubernetes/kubernetes/pull/23707#issuecomment-204141293), are you ok with cherrypicking your unittest fix to the 1.2 branch?

**Comment by bgrant0607** [comment]
@k8s-bot unit test this issue: #IGNORE

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
@bprashanth We don't ask for individual permission to copy commits from master to release branches. I'm going to human-approve the CLA.

**Comment by bgrant0607** [comment]
@k8s-bot unit test this issue: #IGNORE

**Comment by bgrant0607** [comment]
Unit tests fail. PTAL, but this won't make 1.2.1 at this point.

**Comment by bprashanth** [comment]
Got distracted debugging something else. unittests failed because I need to update api ref docs but running update-api-reference.sh (both on master+cherrypick and upstream/release-1.2) didn't fix anything locally, so digging into the script now.

**Comment by bprashanth** [comment]
@caesarxuchao I think there's something fishy with the update scripts. I ran it against both master + this cherrypick and upstream/release-1.2, it finished successfully, but the "verify-api-ref" script failed. Eventually I ran "verify-api-reference-docs.sh" with the rm line commented out, and copied the files from _tmp over the real ones. This got tests to pass. Maybe bug?

**Comment by bprashanth** [comment]
@bgrant0607 PTAL so we can get this in the next minor release

**Comment by bgrant0607** [comment]
Thanks @bprashanth. All green. Merging.

File: hack/verify-api-reference-docs.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ TMP_ROOT="${KUBE_ROOT}/_tmp"
 echo "diffing ${API_REFERENCE_DOCS_ROOT} against freshly generated docs"
 ret=0
 diff -NauprB -I 'Last update' --exclude=*.md "${API_REFERENCE_DOCS_ROOT}" "${OUTPUT_DIR}" || ret=$?
-rm -rf "${TMP_ROOT}"
+# rm -rf "${TMP_ROOT}"
```
Comment (caesarxuchao) [inline hack/verify-api-reference-docs.sh]: I'll add this line back.

I couldn't quickly figure out why the verification failed. I'll help debug when this happens again.

Comment (bprashanth) [inline hack/verify-api-reference-docs.sh]: Ack. didn't mean to comment it out. I assume you're adding it back, let me know if you want me to instead.

**Pull request merged**